File: application/src/main/java/com/catalog/application/category/delete/DefaultDeleteCategoryUseCase.java
package com.catalog.application.category.delete;

import com.catalog.domain.category.CategoryGateway;
import com.catalog.domain.category.CategoryID;

import java.util.Objects;

public class DefaultDeleteCategoryUseCase extends DeleteCategoryUseCase {
    private final CategoryGateway categoryGateway;

    public DefaultDeleteCategoryUseCase(final CategoryGateway categoryGateway) {
        this.categoryGateway = Objects.requireNonNull(categoryGateway);
    }

    @Override
    public void execute(final String anIn) {
        this.categoryGateway.deleteById(CategoryID.from(anIn));
    }
}

File: application/src/main/kotlin/category/delete/DefaultDeleteCategoryUseCase.kt
package com.full.admin.catalog.application.category.delete

import category.CategoryGateway
import category.CategoryID

class DefaultDeleteCategoryUseCase(private val categoryGateway: CategoryGateway) : DeleteCategoryUseCase() {
    override fun execute(anId: String) {
        this.categoryGateway.deleteById(CategoryID.from(anId))
    }
}